src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete cloneState[key];
+        }
+        break;
+
+      case 'clear':
+        cloneState = {};
+        break;
+
+      default:
+        break;
```

Comment:
You can throw an error if there is no such type
```suggestion
      default:
        throw new Error('There is no type like that!')
```